src/types.ts

```diff
@@ -1,3 +1,64 @@
-export type Request = chrome.devtools.network.Request | chrome.devtools.network.HAREntry
+export type Header = { name: string; value: string };
 
-export const isChromeRequest = (request: Request) : request is chrome.devtools.network.Request => (typeof (request as chrome.devtools.network.Request).getContent === 'function')
\ No newline at end of file
+export interface RequestData {
+  url: string;
+  postData?: {
+    text?: string;
+  };
+  headers: Header[];
+}
+
+export interface ResponseContent {
+  text?: string;
+  encoding?: string;
+}
+
+export interface ResponseData {
+  content: ResponseContent;
+}
+
+export interface BaseRequest {
+  request: RequestData;
+  response: ResponseData;
+}
+
+export interface ChromeRequest extends BaseRequest {
+  getContent: (callback: (content: string, encoding: string) => void) => void;
+}
+
+export type Request = ChromeRequest | BaseRequest;
+
+export const isChromeRequest = (request: Request): request is ChromeRequest => {
+  if ('getContent' in request) {
+    return typeof request.getContent === 'function';
+  }
+  return false;
+};
+
+// HAR Log structure for network API
+export interface HARLog {
+  log: {
+    entries: Request[];
+    version: string;
+    creator: {
+      name: string;
+      version: string;
+    };
+    browser?: {
+      name: string;
+      version: string;
+    };
+    pages?: unknown[];
+    comment?: string;
+  };
+}
+
+// Network API interface that works for both Chrome and Firefox
+export interface NetworkAPI {
```

Comment:
this looks almost identical to the Chrome version above, do we need two separate ones?